CSharpSourceCode/Abilities/AbilityManagerMissionLogic.cs

```diff
@@ -42,5 +89,46 @@ public override void OnAgentCreated(Agent agent)
                 }
             }
         }
+
+        private void EnableSpellMode()
+        {
+            abilityComponent.EnableAbilityMode();
+            ChangeKeyBindings();
+            mainHand = Agent.Main.GetWieldedItemIndex(Agent.HandIndex.MainHand);
+            offHand = Agent.Main.GetWieldedItemIndex(Agent.HandIndex.OffHand);
+            Agent.Main.TryToSheathWeaponInHand(Agent.HandIndex.MainHand, Agent.WeaponWieldActionType.WithAnimationUninterruptible);
+            sheathTimer.Start();
+        }
+
+        private void DisableSpellMode()
+        {
+            abilityComponent.DisableAbilityMode();
+            ChangeKeyBindings();
+            Agent.Main.TryToWieldWeaponInSlot(mainHand, Agent.WeaponWieldActionType.WithAnimationUninterruptible, false);
+            Agent.Main.TryToWieldWeaponInSlot(offHand, Agent.WeaponWieldActionType.WithAnimationUninterruptible, false);
+        }
+
+        private void ChangeKeyBindings()
+        {
+            if (abilityComponent.IsAbilityModeOn)
+            {
+                keyContext.GetGameKey(11).KeyboardKey.ChangeKey(InputKey.Invalid);
```

Comment:
we should maybe already a create a issue for this... this will definitely be problems in future... that is a direct reference which could be very problematic... I don't see a problem for the pr now, but we should tackle this soon we can maybe utilize the input system on our own

Reply:
1.6.1 has a whole new system for custom keybinds. We should wait with this for that.

Reply:
yeah right. maybe 1.6.3 or .1.6.4. will be finnally our jump ^^'

Reply:
You're right, will be waiting for 1.6.1